src/main/scala/org/broadinstitute/hail/driver/Grep.scala

```diff
@@ -34,11 +34,19 @@ object Grep extends Command {
     val regex = args.head.r
     val files = args.tail
 
-    sc.union(files.map { f =>
-      sc.textFile(f)
-    }).filter(line => regex.findFirstIn(line).isDefined)
+    sc.textFilesLines(files.toArray)
+      .filter(line => regex.findFirstIn(line.value).isDefined)
       .take(options.max)
-      .foreach(println)
+      .groupBy(_.source.asInstanceOf[TextContext].file)
+      .foreach { case (file, lines) =>
+        println(s"$file:")
+        log.info(s"$file:")
+        lines.foreach { line =>
+          val lineToPrint = truncate(line.value, 60)
+          log.info("\t" + line.value)
+          println(s"\t$lineToPrint")
+        }
```

Comment:
I tried that, but the lines looked pretty cluttered.  If you prefer it, I can add it back in.